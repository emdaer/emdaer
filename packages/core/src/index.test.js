/* eslint-disable import/no-unresolved, import/no-extraneous-dependencies */

jest.mock('@emdaer/plugin-foo', () => async () => 'Hello, World', {
  virtual: true,
});

jest.mock(
  '@emdaer/transform-bar',
  () => async content => content.replace('Hello, World', 'Hello, World!'),
  {
    virtual: true,
  }
);

const emdaer = require('./');

describe('@emdaer/core', () => {
  it('processes an .emdaer.md file', async () => {
    expect(
      await emdaer(
        './.emdaer/README.emdaer.md',
        `# <!--emdaer-p
  - '@emdaer/plugin-foo'
  - a: 0
    b: 1
    c: 2
-->

<!--emdaer-t
  - '@emdaer/transform-bar'
-->
`
      )
    ).toMatchSnapshot();
  });
});
