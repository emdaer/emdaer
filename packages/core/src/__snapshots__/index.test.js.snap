// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emdaer/core processes an .emdaer.md file 1`] = `
"<h1 id=\\"hello-world-\\">Hello, World!</h1>
"
`;
